### PR TITLE
Add .rgb properties + tests for Color, Sprite, and BasicSprite

### LINKED ISSUE #1838: Make Color constants usable without affecting alpha
### What should be added/changed?

Add an "rgb" method to arcade.types.Color which return a 3-tuple (type "RGB") based on the color, i.e., it leaves out the alpha.

### What would it help with?

In arcade 2, Sprite.alpha was separate from Sprite.color, so one could use the constants in arcade.color to set the color without changing the alpha of the sprite. For example, `self.color = arcade.color.WHITE` would not affect the alpha of the Sprite. Now it does, which means makes the color constants much less usable. With an "rgb" property, one could do `self.color = arcade.color.WHITE.rgb` which would not affect alpha since the color setter checks the length of the input.

One can currently do the same, of course, by either inputting the color numbers directly as a tuple (but then what are the color constants even good for?) or by grabbing the constant and accessing its r, g, and b properties (but that is very messy).

```
Arcade 3.0.0.dev23
------------------
vendor: Intel
renderer: Intel(R) UHD Graphics
version: (3, 3)
python: 3.11.0 (main, Oct 24 2022, 18:26:48) [MSC v.1933 64 bit (AMD64)]
platform: win32
pyglet version: 2.0.7
PIL version: 9.4.0
```

### PR DESCRIPTION
**Pull request by pushfoo**
#### To the old PR:

Supersedes #1970 by:
* Resolving merge conflicts
* Squashing down the old stalled PR into one commit (FriendlyGecko still gets credit!)
* Reverting non-critical changes

#### To Arcade

Closes #1838 ([Marked as 3.0 Mandatory](https://github.com/pythonarcade/arcade/milestone/8)) by:

* Add `.rgb` syntactic sugar property to `Color`
* Add `.rgb` property to `BasicSprite` and subclasses:
  * It largely works like `BasicSprite.color`
  * The difference is it always ignores any present alpha value
* Add preliminary tests

The groundwork for this was handled by #2029.

#### Remaining questions

* Add exhaustive tests for channel setting? (these will be iterative, not unrolled)
* Add similar tests for `.color`?

### CODE REVIEW

**Comment by pushfoo** [comment]
The *GitHub basd tests / Code inspections* step is failing on GUI / Text pyright issues as it has been recently.

**Comment by pushfoo** [comment]
TL;DR: This is ready to review.

The recent force-push is a rebase onto development to get #2062's pyright fixes.

**Comment by FriendlyGecko** [comment]
In regards to my previous merge request, should I close it or is there something still relevant about it?

**Comment by pushfoo** [comment]
@FriendlyGecko TL;DR: Don't delete the branch yet.

We split the types module recently in #2065. That makes it easier to do future color improvements, but they'll probably have to wait till 3.1+ at the earliest.

Before we do more with color, we first have a bunch of other things to do:
* 3.0 fixes for cameras, including the rectangle cleanup DigiDuncan has started
* Exploring a `Lerpable` [`Protocol` type](https://docs.python.org/3.8/library/typing.html#typing.Protocol)
* Cleaning up the math module

After 3.0, we might be able to start improving color types further.
